Library/Homebrew/brew.rb

```diff
@@ -166,35 +168,35 @@
       Try to figure out the problem yourself and submit a fix as a pull request.
       We will review it but may or may not accept it.
     EOS
+
   end
 
   exit 1
-rescue RuntimeError, SystemCallError => e
-  raise if e.message.empty?
+rescue Exception => e # rubocop:disable Lint/RescueException
+  runtime_or_system_call_error = e.is_a?(RuntimeError) || e.is_a?(SystemCallError)
+  raise if runtime_or_system_call_error && e.message.empty?
 
   onoe e
-  $stderr.puts Utils::Backtrace.clean(e) if args&.debug? || ARGV.include?("--debug")
 
-  exit 1
-rescue MethodDeprecatedError => e
-  onoe e
-  if e.issues_url
-    $stderr.puts "If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):"
-    $stderr.puts "  #{Formatter.url(e.issues_url)}"
+  method_deprecated_error = e.is_a?(MethodDeprecatedError)
+  runtime_or_system_call_or_method_deprecated_error = runtime_or_system_call_error || method_deprecated_error
+  if args&.debug? || ARGV.include?("--debug") || !runtime_or_system_call_or_method_deprecated_error
+    $stderr.puts Utils::Backtrace.clean(e)
   end
-  $stderr.puts Utils::Backtrace.clean(e) if args&.debug? || ARGV.include?("--debug")
-  exit 1
-rescue Exception => e # rubocop:disable Lint/RescueException
-  onoe e
-  if internal_cmd && !OS.unsupported_configuration?
-    if Homebrew::EnvConfig.no_auto_update?
-      $stderr.puts "#{Tty.bold}Do not report this issue until you've run `brew update` and tried again.#{Tty.reset}"
-    else
-      $stderr.puts "#{Tty.bold}Please report this issue:#{Tty.reset}"
-      $stderr.puts "  #{Formatter.url(OS::ISSUES_URL)}"
-    end
+
+  if OS.unsupported_configuration?
+    $stderr.puts "#{Tty.bold}Do not report this issue: you are running in an unsupported configuration.#{Tty.reset}"
+  elsif Homebrew::EnvConfig.no_auto_update?
+    $stderr.puts "#{Tty.bold}You have disabled automatic updates.#{Tty.reset}"
+    $stderr.puts "#{Tty.bold}Do not report this issue until you've run `brew update` and tried again.#{Tty.reset}"
```

Comment:
**suggestion (non-blocking):** 
```suggestion
    $stderr.puts "#{Tty.bold}You have disabled automatic updates.#{Tty.reset}"
    $stderr.puts "#{Tty.bold}Do not report this issue until you've run `brew update` and tried again or unset HOMEBREW_NO_AUTO_UPDATE.#{Tty.reset}"
```

Soft pitch on the instruction to clear the envvar…